Firestore/core/test/unit/api/load_bundle_task_test.cc

```diff
@@ -0,0 +1,415 @@
+                                     FAIL() << "Removed observer is called.";
+                                   });
+  task.RemoveObserver(handle1);
+
+  auto handle2 = task.ObserveState(LoadBundleTaskState::Success,
+                                   [&](LoadBundleTaskProgress p) {
+                                     EXPECT_EQ(p, SuccessProgress());
+                                     queue.push(1);
+                                   });
+
+  task.SetSuccess(SuccessProgress());
+
+  EXPECT_EQ(1, queue.pop());
+}
+
+TEST(LoadBundleTaskTest, ErrorObserveTriggers) {
+  auto error_progress = ErrorProgress();
+  LoadBundleTask task(CreateUserQueue());
+  BlockingQueue<int> queue;
+
+  auto handle1 = task.ObserveState(LoadBundleTaskState::Error,
+                                   [&](LoadBundleTaskProgress p) {
+                                     EXPECT_EQ(p, error_progress);
+                                     queue.push(1);
+                                   });
+  auto handle2 = task.ObserveState(LoadBundleTaskState::Error,
+                                   [&](LoadBundleTaskProgress p) {
+                                     EXPECT_EQ(p, error_progress);
+                                     (void)p;
+                                     queue.push(2);
+                                   });
+  auto handle3 = task.ObserveState(LoadBundleTaskState::Error,
+                                   [&](LoadBundleTaskProgress p) {
+                                     EXPECT_EQ(p, error_progress);
+                                     queue.push(3);
+                                   });
+
+  task.SetError();
+
+  EXPECT_EQ(1, queue.pop());
+  EXPECT_EQ(2, queue.pop());
+  EXPECT_EQ(3, queue.pop());
+}
+
+TEST(LoadBundleTaskTest, RemovesObserverByState) {
+  LoadBundleTask task(CreateUserQueue());
+
+  auto handle1 = task.ObserveState(LoadBundleTaskState::Error,
+                                   [&](LoadBundleTaskProgress p) {
+                                     (void)p;
+                                     FAIL() << "Removed observer is called.";
+                                   });
+  task.RemoveObservers(LoadBundleTaskState::Error);
+  task.SetError();
+
+  BlockingQueue<int> queue;
+  auto error_progress = ErrorProgress();
+  auto handle2 = task.ObserveState(LoadBundleTaskState::Error,
+                                   [&](LoadBundleTaskProgress p) {
+                                     EXPECT_EQ(p, error_progress);
+                                     queue.push(1);
+                                   });
+
+  task.SetError();
+
+  EXPECT_EQ(1, queue.pop());
+}
+
+TEST(LoadBundleTaskTest, ProgressObserveTriggers) {
+  LoadBundleTask task(CreateUserQueue());
+  BlockingQueue<int> queue;
+
+  auto handle1 = task.ObserveState(LoadBundleTaskState::InProgress,
+                                   [&](LoadBundleTaskProgress p) {
+                                     EXPECT_EQ(p, Progress(1, 5));
```

Comment:
Nit: consider storing `Progress(1, 5)` in a local variable.

Reply:
Done.